hyperspace/core/src/chain.rs

```diff
@@ -177,20 +192,28 @@ impl IbcProvider for AnyChain {
 	{
 		match self {
 			AnyChain::Parachain(chain) => {
-				let finality_event = ibc::downcast!(finality_event => AnyFinalityEvent::Parachain)
+				let finality_event = downcast!(finality_event => AnyFinalityEvent::Parachain)
+					.ok_or_else(|| AnyError::Other("Invalid finality event type".to_owned()))?;
+				let (client_msg, events, update_type) =
+					chain.query_latest_ibc_events(finality_event, counterparty).await?;
+				Ok((client_msg, events, update_type))
+			},
+			#[cfg(feature = "cosmos")]
+			AnyChain::Cosmos(chain) => {
+				let finality_event = downcast!(finality_event => AnyFinalityEvent::Cosmos)
 					.ok_or_else(|| AnyError::Other("Invalid finality event type".to_owned()))?;
 				let (client_msg, events, update_type) =
 					chain.query_latest_ibc_events(finality_event, counterparty).await?;
 				Ok((client_msg, events, update_type))
 			},
-			_ => unreachable!(),
 		}
 	}
 
 	async fn ibc_events(&self) -> Pin<Box<dyn Stream<Item = IbcEvent> + Send + 'static>> {
 		match self {
 			Self::Parachain(chain) => chain.ibc_events().await,
-			_ => unreachable!(),
+			#[cfg(feature = "cosmos")]
```

Comment:
Why feature gating?

Reply:
I just thought that it's better to have it than not when it will be needed :) The reason is that when hyperspace supports many clients, the end user will want to opt out of some of them to increase compilation times or not waste time installing other chain dependencies.